src/outbackcdx/ChangePollingThread.java

```diff
@@ -0,0 +1,121 @@
+                System.out.println("Received I/O exception while processing " + primaryReplicationUrl);
+                e.printStackTrace();
+            } catch (RocksDBException e){
+                System.out.println("The plane has crashed into the mountain. RocksDB threw an exception during replication from "+ primaryReplicationUrl);
+                e.printStackTrace();
+            } catch (Exception e) {
+                System.out.println("Dang! something happened while processing " + primaryReplicationUrl);
+                e.printStackTrace();
+            }
+
+            long sleepTime = (pollingInterval * 1000) - (System.currentTimeMillis() - startTime);
+            if (sleepTime > 0) {
+                try {
+                    Thread.sleep(sleepTime);
+                } catch (InterruptedException e) {
+                    System.out.println("Received interruption at " + System.currentTimeMillis());
+                    return;
+                }
+            }
+        }
+    }
+
+    private void replicate() throws IOException, RocksDBException {
+        // strip trailing slash if necessary
+        CloseableHttpClient httpclient = HttpClients.createDefault();
+        HttpGet request = new HttpGet(primaryReplicationUrl);
+        long sequenceNumber = 0;
+        String writeBatch = null;
+        HttpResponse response = httpclient.execute(request);
+
+        if(response.getStatusLine().getStatusCode() != 200){
+            InputStream inputStream = response.getEntity().getContent();
+            String contentString = new BufferedReader(new InputStreamReader(inputStream)).readLine();
+            throw new IOException("Received '" + response.getStatusLine() + "' response from " + primaryReplicationUrl + ": \n" + contentString);
+        }
+        InputStream content = response.getEntity().getContent();
+        JsonReader reader = GSON.newJsonReader(new InputStreamReader(content));
+        reader.beginArray();
+        while (reader.hasNext()) {
+            JsonToken nextToken = reader.peek();
+            if (JsonToken.BEGIN_OBJECT.equals(nextToken)) {
+                reader.beginObject();
+            } else if(JsonToken.NAME.equals(nextToken)){
+                String name  =  reader.nextName();
+                if(name.equals("sequenceNumber")){
+                    sequenceNumber = Long.valueOf(reader.nextString());
+                } else if (name.equals("writeBatch")){
+                    writeBatch = reader.nextString();
+                }
+            } else if (JsonToken.END_OBJECT.equals(nextToken)){
+                commitWriteBatch(index, sequenceNumber, writeBatch);
+            }
+        }
+    }
+
+    private void commitWriteBatch(Index index, long sequenceNumber, String writeBatch) throws RocksDBException {
+        Base64.Decoder decoder = Base64.getDecoder();
+        byte[] decodedWriteBatch = decoder.decode(writeBatch);
+        WriteBatch batch = new WriteBatch(decodedWriteBatch);
```

Comment:
WriteBatch like most of the RocksDB API should be closed as it wraps a C++ object which has manual memory management. Usually the try-with-resources statement is the way to go (to properly handle exceptions):

```java
try (WriteBatch batch = new WriteBatch(...)) {
    ...
}
```